File: api/src/main/java/net/onelitefeather/pandorascluster/dbo/land/LandDBO.java
package net.onelitefeather.pandorascluster.dbo.land;

import net.onelitefeather.pandorascluster.dbo.player.LandPlayerDBO;
import net.onelitefeather.pandorascluster.dbo.position.HomePositionDBO;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public interface LandDBO {
    @Nullable
    Long id();

    @NotNull
    LandPlayerDBO owner();

    @NotNull
    HomePositionDBO home();

    @NotNull
    List<LandAreaDBO> areas();

    @NotNull
    String world();
}

File: api/src/main/kotlin/net/onelitefeather/pandorascluster/dbo/land/LandDBO.kt
package net.onelitefeather.pandorascluster.dbo.land

import net.onelitefeather.pandorascluster.dbo.chunk.ClaimedChunkDBO
import net.onelitefeather.pandorascluster.dbo.flag.FlagRoleAttachmentDBO
import net.onelitefeather.pandorascluster.dbo.player.LandMemberDBO
import net.onelitefeather.pandorascluster.dbo.player.LandPlayerDBO
import net.onelitefeather.pandorascluster.dbo.position.HomePositionDBO

interface LandDBO {

    fun id(): Long?

    fun owner(): LandPlayerDBO

    fun home(): HomePositionDBO

    fun members(): List<LandMemberDBO>

    fun chunks(): List<ClaimedChunkDBO>

    fun flags(): List<FlagRoleAttachmentDBO>

    fun world(): String
}